float32 futek1
float32 futek2
